### PR TITLE
Support prefix messages (e.g. system prompt) in chat engine and OpenAI agent

### PR DESCRIPTION
**Pull request by Disiok**
# Description
Support prefix messages (e.g. system prompt) in chat engine and OpenAI agent
Add a bank of system prompts taken from Azure OpenAI Studio (with attribution) 

### Caveat
Not supported in condense question mode yet.

### CODE REVIEW

File: docs/examples/chat_engine/chat_engine_personality.ipynb
Comment (logan-markewich) [inline docs/examples/chat_engine/chat_engine_personality.ipynb:1]: I think some cell outputs are missing here (love the examples though lol)

File: llama_index/agent/context_retriever_agent.py [hunk 1]
```diff
@@ -59,13 +57,15 @@ def __init__(
         context_separator: str,
         llm: OpenAI,
         chat_history: List[ChatMessage],
+        prefix_messages: List[ChatMessage],
```
Comment (logan-markewich) [inline llama_index/agent/context_retriever_agent.py:60]: should we update the doctoring?

File: llama_index/prompts/system.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+SHAKESPEARE_WRITING_ASSISTANT = """\
+You are a Shakespearean writing assistant who speaks in a Shakespearean style. You help people come up with creative ideas and content like stories, poems, and songs that use Shakespearean style of writing style, including words like "thou" and "hath”.
+Here are some example of Shakespeare's style:
+ - Romeo, Romeo! Wherefore art thou Romeo?
+ - Love looks not with the eyes, but with the mind; and therefore is winged Cupid painted blind.
+ - Shall I compare thee to a summer's day? Thou art more lovely and more temperate.
+"""
+
+IRS_TAX_CHATBOT = """\
```
Comment (logan-markewich) [inline llama_index/prompts/system.py:11]: Lol this is so specific. I love it.

File: llama_index/agent/openai_agent.py [hunk 3]
```diff
@@ -142,12 +134,14 @@ def __init__(
...
         llm: OpenAI,
         chat_history: List[ChatMessage],
+        prefix_messages: Optional[List[ChatMessage]],
         verbose: bool = False,
         max_function_calls: int = DEFAULT_MAX_FUNCTION_CALLS,
         callback_manager: Optional[CallbackManager] = None,
     ) -> None:
         self._llm = llm
         self._chat_history = chat_history
+        self._prefix_messages = prefix_messages
```
Comment (logan-markewich) [inline llama_index/agent/openai_agent.py:144]: This isn't actually used in the chat functions?

Comment (logan-markewich) [reply llama_index/agent/openai_agent.py:144]: Might want to add it to `init_chat` for now? Will need some extra thought once we have memory modules

Comment (Disiok) [reply llama_index/agent/openai_agent.py:144]: ooops, missing commits here.

**Pull request merged**